### PR TITLE
expose seqno from compaction filter and iterator

### PR DESCRIPTION
**Pull request by hicqu**
This PR supports to access `seqno` for every key/value pairs in compaction filter or iterator.
It's helpful to enhance GC in compaction filter in TiKV.

### CODE REVIEW

File: db/db_iter.cc [hunk 1]
```diff
@@ -195,6 +195,10 @@ class DBIter final: public Iterator {
       return iter_.value();
     }
   }
+  bool seqno(SequenceNumber* no) const override {
+    *no = ikey_.sequence;
```
Comment (yiwu-arbug) [inline db/db_iter.cc:199]: `assert(valid_);`

File: include/rocksdb/iterator.h [hunk 2]
```diff
@@ -78,6 +79,11 @@ class Iterator : public Cleanable {
   // REQUIRES: Valid()
   virtual Slice value() const = 0;
 
+  // Return the sequence number for the current entry if it's available.
+  // Return false if it's not available.
+  // REQUIRES: Valid()
+  virtual bool seqno(SequenceNumber* /*seqno*/) const { return false; }
```
Comment (yiwu-arbug) [inline include/rocksdb/iterator.h:85]: do you need to update titan to support this method?

Comment (hicqu) [reply include/rocksdb/iterator.h:85]: It's not necessary because now we only use it for write CF, which its value is very small.

**Review by Connor1996 — APPROVED** [review]
LGTM, though this API is weird for others.🤣

File: include/rocksdb/iterator.h (hunk 2 above)
Comment (yiwu-arbug) [reply include/rocksdb/iterator.h:85]: Let's still update Titan. It sounds like a 坑 in the future.

**Review by yiwu-arbug — APPROVED** [review]
LGTM

**Pull request merged**